### ISSUE TITLE
Cilium CNI with multiple nodes fails

### ISSUE THREAD

**Issue by code-merc** [kubernetes/minikube#11391]
**Steps to reproduce the issue:** 

1. Run
```bash
 minikube start --cni=cilium --driver=docker --nodes=2  --addons=default-storageclass,registry,storage-provisioner
```
Then run this command so I can connect to the registry on my local machine
```bash
docker run --rm -it --network=host alpine ash -c "apk add socat && socat TCP-LISTEN:5000,reuseaddr,fork TCP:$(minikube ip):5000"
```
Sometimes I can push to the registry, sometimes I get errors in the alpine container like 
```
2021/05/12 21:01:43 socat[13] E connect(5, AF=2 192.168.49.2:5000, 16): Connection refused
``` 
When I'm able to push to the registry I have not been able to get an image to successfully pull, I'll get connection refused errors and an `ErrImagePull` for the pod. I don't have this same issue when running with a single node, although I have had issues with `kind` and `minikube` cluster creation using the default CNI after tearing down the cilium cluster. The the only way to get clusters to successfully start after that is to reboot my machine, otherwise I'll get errors that look like:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox : failed to set bridge addr: could not add IP address to "cni0": permission denied
```
So seems like cilium does something to kindnet even after the cluster has been deleted.

**Output of `minikube logs`** 
[logs.txt](https://github.com/kubernetes/minikube/files/6469083/logs.txt)

**Comment by code-merc** [comment]
To remove any possible issues with EBPF and my host machine I used the `kvm2` driver by running:

```
minikube start --driver=kvm2 --nodes=2 --cni=cilium --addons=default-storageclass,registry,storage-provisioner
```

Applying the cilium test with
```
> kubectl create ns cilium-test

> kubectl apply --namespace=cilium-test -f https://raw.githubusercontent.com/cilium/cilium/v1.7/examples/kubernetes/connectivity-check/connectivity-check.yaml

> kubectl get pods -n cilium-test

NAME                                                     READY   STATUS             RESTARTS   AGE
echo-a-8b6595b89-h8r5r                                   1/1     Running            0          15m
echo-b-57d8cc56dc-p5rr7                                  1/1     Running            0          15m
echo-b-host-7bccc85c7c-n4zp7                             1/1     Running            0          15m
host-to-b-multi-node-clusterip-7b87db6659-286bb          1/1     Running            0          15m
host-to-b-multi-node-headless-8b4797fd9-wwjcb            1/1     Running            1          15m
pod-to-a-5567c85856-c4qvc                                1/1     Running            0          15m
pod-to-a-allowed-cnp-7b85c8db8-f4r67                     1/1     Running            0          15m
pod-to-a-external-1111-77b455b8bb-58lhn                  1/1     Running            0          15m
pod-to-a-l3-denied-cnp-7f64d7b7c4-dkr79                  1/1     Running            0          15m
pod-to-b-intra-node-6f68b7c94f-g4gks                     1/1     Running            0          15m
pod-to-b-intra-node-nodeport-7f77495d64-nmpnk            0/1     CrashLoopBackOff   7          15m
pod-to-b-multi-node-clusterip-9c4b88cc7-qbj4p            1/1     Running            0          15m
pod-to-b-multi-node-headless-757985b7bf-5bzmh            1/1     Running            1          15m
pod-to-b-multi-node-nodeport-666566f6c8-zqc2t            0/1     CrashLoopBackOff   7          15m
pod-to-external-fqdn-allow-google-cnp-6557d6b7d6-cw5qm   1/1     Running            0          14m
```
Some tests are failing, and when I forward to the registry with `kubectl port-forward --namespace kube-system registry-8hcnb 5000:5000` I can push successfully, but pods will fail with `ImagePullBackoff` and `Failed to pull image "localhost:5000/example-html-image": rpc error: code = Unknown desc = Error response from daemon: Get http://localhost:5000/v2/: dial tcp 127.0.0.1:5000: connect: connection refused`

**Comment by sharifelgamal** [comment]
I believe this has been fixed with https://github.com/kubernetes/minikube/pull/12587, could you try again and report back?

**Comment by spowelljr** [comment]
Hi @code-merc, we haven't heard back from you, if you have a chance please try this again with the latest version of minikube. For now I'm going to assume this is fixed by #12587, if it's not feel free to reopen this issue again, thanks!
